TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -755,6 +755,95 @@ public override bool Visit(DataDefinition dataDefinition)
+                    dataEntry.Picture != null)
+                {
+                    DiagnosticUtils.AddError(dataDefinition,
+                        "Variable with usage COMP-1 and COMP-2 cannot have a PICTURE", dataEntry);
+                }
+
+                if (dataEntry?.LevelNumber != null && IsDataDefinitionEmpty(dataDefinition, dataEntry))
+                {
+                    //Get current node index
+                    var nodeIndex = dataDefinition.Parent.IndexOf(dataDefinition);
+                    //Get sibling nodes
+                    var siblingNodes = dataDefinition.Parent.Children;
+                    //Get immediately following DataDefinition
+                    var nextData = siblingNodes.Skip(nodeIndex + 1).OfType<DataDefinition>().FirstOrDefault();
+                    if (nextData != null && nextData.IsInsideCopy())
+                    {
+                        DiagnosticUtils.AddError(dataDefinition, $"Cannot include copy {nextData.CodeElement.FirstCopyDirective.TextName} " +
+                                                                  $"under level {dataEntry.LevelNumber} " +
+                                                                  $"because copy starts at level {nextData.CodeElement.LevelNumber}.", dataEntry);
+                    }
+                    else
+                    {
+                        DiagnosticUtils.AddError(dataDefinition, "A group item cannot be empty. Add children, picture or usage declaration.", dataEntry);
+                    }
+                }
+            }
+        }
+
+        private static bool IsDataDefinitionEmpty([NotNull] DataDefinition dataDefinition, [NotNull] CommonDataDescriptionAndDataRedefines dataEntry)
+        {
+            if (dataDefinition.IsTableOccurence)
+            {
+                // OCCURS: not empty if PICTURE or type definition
+                //TODO We systematically throw an error in case of empty array with a TYPEDEF defined on parent group
+                //This could be improved by checking after type expansion
+                if ((dataEntry.Picture != null) || (dataDefinition.TypeDefinition != null))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                // Others: exclude RENAME + not empty if PICTURE or user defined data type
+                if (dataDefinition.CodeElement.LevelNumber?.Value >= 50 || dataDefinition.Picture != null || dataEntry.UserDefinedDataType != null)
+                {
+                    return false;
+                }
+            }
+
+            if (dataDefinition.Usage.HasValue && IsUsageAllowed(dataDefinition.Usage.Value))
+            {
+                // Empty because usage is not allowed
+                return false;
+            }
+
+            // Not empty if at least one child (but ignore Index chidren)
+            return dataDefinition.Children.Count(c => c is not IndexDefinition) == 0;
```

Comment:
Children to exclude can also include level 88 and renames. I forgot to test theses cases.
I will update test in the issue.

Reply:
Test file = `Level88AndUsage.rdzb.cbl`